graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/util/KeyUtil.java

```diff
@@ -0,0 +1,173 @@
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KeyUtil {
+    private static final Logger LOG = LoggerFactory.getLogger(KeyUtil.class);
+    private static final Joiner JOINER = Joiner.on(",").skipNulls();
+    private static final Pattern KEY_PATTERN = Pattern.compile("-{5}BEGIN (?:(RSA|DSA)? )?(ENCRYPTED )?PRIVATE KEY-{5}\\r?\\n([A-Z0-9a-z+/\\r\\n]+={0,2})\\r?\\n-{5}END (?:(?:RSA|DSA)? )?(?:ENCRYPTED )?PRIVATE KEY-{5}\\r?\\n$", Pattern.MULTILINE);
+
+    public static TrustManager[] initTrustStore(File tlsClientAuthCertFile)
+            throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        final KeyStore trustStore = KeyStore.getInstance("JKS");
+        trustStore.load(null, null);
+        loadCertificates(trustStore, tlsClientAuthCertFile, CertificateFactory.getInstance("X.509"));
+
+        LOG.info("TrustStore: {}, aliases: {}", trustStore, join(trustStore.aliases()));
+
+        final TrustManagerFactory instance = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        instance.init(trustStore);
+
+        return instance.getTrustManagers();
+    }
+
+    @VisibleForTesting
+    protected static void loadCertificates(KeyStore trustStore, File certFile, CertificateFactory cf)
+            throws CertificateException, KeyStoreException, IOException {
+        if (certFile.isFile()) {
+            final Collection<? extends Certificate> certificates = cf.generateCertificates(new FileInputStream(certFile));
+            int i = 0;
+            for (Certificate cert : certificates) {
+                final String alias = certFile.getAbsolutePath() + "_" + i;
+                trustStore.setCertificateEntry(alias, cert);
+                i++;
+                LOG.debug("Added certificate with alias {} to trust store: {}", alias, cert);
+            }
+        } else if (certFile.isDirectory()) {
+            for (Path f : Files.newDirectoryStream(certFile.toPath())) {
+                loadCertificates(trustStore, f.toFile(), cf);
+            }
+        }
+    }
+
+    public static KeyManager[] initKeyStore(File tlsKeyFile, File tlsCertFile, String tlsKeyPassword)
+            throws IOException, GeneralSecurityException {
+        final KeyStore ks = KeyStore.getInstance("JKS");
+        ks.load(null, null);
+        final CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        final Collection<? extends Certificate> certChain = cf.generateCertificates(new FileInputStream(tlsCertFile));
+        final PrivateKey privateKey = loadPrivateKey(tlsKeyFile, tlsKeyPassword);
+        final char[] password = Strings.nullToEmpty(tlsKeyPassword).toCharArray();
+        ks.setKeyEntry("key", privateKey, password, certChain.toArray(new Certificate[certChain.size()]));
+
+        LOG.info("KeyStore: {}, aliases: {}", ks, join(ks.aliases()));
```

Comment:
it has to be more expensive, if client certificates are involved, it simply has to parse more files.

live replacing could be achieved by filesystem watchers, e.g.